dapps/digital-twin-detail/src/components/overview/DetailOverview.ts

```diff
@@ -17,13 +17,103 @@
   the following URL: https://evan.network/license/
 */
 
-// vue imports
 import Component, { mixins } from 'vue-class-component';
-
-// evan.network imports
 import { EvanComponent } from '@evan.network/ui-vue-core';
+import axios from 'axios';
+import {
+  utils, Runtime, Profile, ProfileOptions,
+} from '@evan.network/api-blockchain-core';
+import {
+  DAppTwin, TransactionsResponse, DigitalTwinResponse, TwinTransaction,
+} from 'core/digital-twin-lib';
+import { bccUtils } from '@evan.network/ui';
 
 @Component
-export default class EvanTwinDetailOverviewComponent extends mixins(EvanComponent) {
+export default class DetailOverviewComponent extends mixins(EvanComponent) {
+  twin: DAppTwin = null;
+
+  runtime: Runtime = null;
+
+  transactions: TwinTransaction[] = null;
+
+  async created(): Promise<void> {
+    this.runtime = this.getRuntime();
+    this.twin = this.$store.state.twin;
+    await this.attachCreatedAt();
+    await this.getLastTransactions();
+  }
+
+  async getLastTransactions(): Promise<void> {
+    const authHeaders = await utils.getSmartAgentAuthHeaders(this.runtime);
+    const core = this.runtime.environment === 'testcore' ? '.test' : '';
+    const url = `https://search${core}.evan.network/api/smart-agents/search`;
+
+    const params = {
+      count: 5,
+      offset: 0,
+      reverse: true,
+      sortBy: 'timestamp',
+      address: this.twin.contractAddress,
+    };
+
+    const { data } = await axios.get<TransactionsResponse>(
+      `${url}/transactions/twin`,
+      {
+        headers: {
+          Authorization: authHeaders,
+        },
+        params,
+      },
+    );
+    this.transactions = await this.enhanceTransactions(data.result);
+  }
+
+  /**
+   * Adds and formats properties for displaying Transactions
+   * @param transactions list of transactions to be transformed
+   */
+  async enhanceTransactions(transactions: TwinTransaction[]): Promise<TwinTransaction[]> {
+    return Promise.all(transactions.map(async (transaction) => {
+      const enhancedTransaction = transaction;
+
+      // Add alias of the initiator if known
+      enhancedTransaction.initiator = await bccUtils.getUserAlias(new Profile({
+        accountId: this.runtime.activeAccount,
+        profileOwner: transaction.from,
+        ...(this.runtime as ProfileOptions),
+      }));
+
+      // Add fee in EVE
+      enhancedTransaction.feeInEve = parseFloat(
+        this.runtime.web3.utils.fromWei(
+          (transaction.gas * parseFloat(transaction.gasPrice)).toString(),
+          'ether',
+        ),
+      ).toFixed(5);
+
+      return enhancedTransaction;
+    }));
+  }
+
+  /**
+   * Enhance the current twin with createdAt timestamp
+   */
+  async attachCreatedAt(): Promise<void> {
+    const authHeaders = await utils.getSmartAgentAuthHeaders(this.runtime);
+    const core = this.runtime.environment === 'testcore' ? '.test' : '';
+    const url = `https://search${core}.evan.network/api/smart-agents/search`;
```

Comment:
why not put all the request logic in a vue-agnostic service? Would also have the benefit of determining the request Url only once on initialization and the component would be way smaller.

Reply:
Done. authheaders can't be in constructor cuz of its async nature.